backtesting/backtesting.py

```diff
@@ -161,44 +161,54 @@ def next(self):
             super().next()
         """
 
-    class __FULL_EQUITY(float):
+    class __FULL_EQUITY(float):  # noqa: N801
         def __repr__(self): return '.9999'
 
     _FULL_EQUITY = __FULL_EQUITY(1 - sys.float_info.epsilon)
 
     def buy(self, *,
             size: float = _FULL_EQUITY,
-            limit: float = None,
-            stop: float = None,
-            sl: float = None,
-            tp: float = None,
+            limit: Optional[float] = None,
+            stop: Optional[float] = None,
+            sl: Optional[float] = None,
+            tp: Optional[float] = None,
             indicator: str = None,
+            tag: object = None
             ):
         """
         Place a new long order. For explanation of parameters, see `Order` and its properties.
 
+        See `Position.close()` and `Trade.close()` for closing existing positions.
+
         See also `Strategy.sell()`.
         """
-        assert 0 < size < 1 or round(size) == size, \
-            "size must be a positive fraction of equity, or a positive whole number of units"
-        return self._broker.new_order(size, limit, stop, sl, tp, indicator)
+        # assert 0 < size < 1 or round(size) == size, \
+        #     "size must be a positive fraction of equity, or a positive whole number of units"
+
+        return self._broker.new_order(size, limit, stop, sl, tp, indicator, tag)
 
     def sell(self, *,
+             # TODO here we can trade adjusted size
              size: float = _FULL_EQUITY,
-             limit: float = None,
-             stop: float = None,
-             sl: float = None,
-             tp: float = None,
+             limit: Optional[float] = None,
+             stop: Optional[float] = None,
+             sl: Optional[float] = None,
+             tp: Optional[float] = None,
              indicator: str = None,
+             tag: object = None
              ):
         """
         Place a new short order. For explanation of parameters, see `Order` and its properties.
 
         See also `Strategy.buy()`.
+
+        .. note::
+            If you merely want to close an existing long position,
+            use `Position.close()` or `Trade.close()`.
         """
-        assert 0 < size < 1 or round(size) == size, \
-            "size must be a positive fraction of equity, or a positive whole number of units"
-        return self._broker.new_order(-size, limit, stop, sl, tp, indicator)
+        # assert 0 < size < 1 or round(size) == size, \
```

Comment:
add explanations pls with links to other places, if we need to fix it again later